src/types.jl

```diff
@@ -858,8 +888,16 @@ AtomsBase.velocity(s::ReplicaSystem, i::Integer) = s.replicas[1].velocities[i]
 AtomsBase.atomic_mass(s::Union{System, ReplicaSystem}) = masses(s)
 AtomsBase.atomic_mass(s::Union{System, ReplicaSystem}, i::Integer) = mass(s.atoms[i])
 
-AtomsBase.atomic_number(s::Union{System, ReplicaSystem}) = fill(missing, length(s))
-AtomsBase.atomic_number(s::Union{System, ReplicaSystem}, i::Integer) = missing
+# Base.getindex(sys::Union{System, ReplicaSystem}, i::Integer, x::Symbol) =
+#     getfield(getindex(sys, i), x)
+function Base.getindex(system::Union{System, ReplicaSystem}, i, x::Symbol)
+    if hasatomkey(system, x)
```

Comment:
In contrast to what I said earlier, it might be better to explicitly list the mandatory atom keys here (the ones you currently put in `atomkeys`) and for `:charge` and similar do something else (e.g. explicitly insert the code that retrieves them).

Reply:
So basically manually add branches to the if statement for keys not in AtomsBase?

Reply:
You could do that or alternatively you could use the trick I proposed only for a whitelisted set of keys (the mandatory ones).

Reply:
Ok cool that worked. Pushed changes.

I'm not the biggest fan of having to manually maintain that list of `custom_keys` (line 886) inside the `getindex` function. Maybe I'm missing a cleaner way to do that but it could also be useful add a `AtomsBase.extra_keys()` or something like that to specify extensions to the interface. This would also let you define the getindex function inside of the AtomsBase library that makes use of this function to define a generic `getidnex `for `AtomView`.

Reply:
Hmm. Not fully sure if that might be over the top, but I'm open to discussing it further.

Certainly we should have programmatic ways to access the mandatory keys and the optional, standardised keys (like charge).

Would youmind opening an issue at atomsbase? I feel this is also an excellent topic for a JuliaCon hackaton. Would you agree?

Reply:
Yea I do, I'll open a PR.